connect/runtime/src/main/java/org/apache/kafka/connect/runtime/AbstractWorkerTask.java

```diff
@@ -0,0 +1,92 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package org.apache.kafka.connect.runtime;
+
+import org.apache.kafka.connect.util.ConnectorTaskId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+abstract class AbstractWorkerTask implements WorkerTask {
```

Comment:
Definitely not critical, but does the `WorkerTask` interface actually buy us anything since there are no new public methods implemented here and this is the only class the implements it? What if we just dropped it, rename `AbstractWorkerTask` just to `WorkerTask` and have it `implements Runnable`? There's sometimes value in having the interface if a) its public API and you might expect new implementations of it or b) the abstract class adds some other public methods such that in some contexts having an `AbstractWorkerTask` may actually differ from having a `WorkerTask`, but those don't apply here.